src/wp-includes/media.php

```diff
@@ -1933,15 +1933,10 @@ function wp_filter_content_tags( $content, $context = null ) {
  * @return string Converted `img` tag with optimization attributes added.
  */
 function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
-	$image_elm = new WP_HTML_Tag_Processor( $image );
-	if ( ! $image_elm->next_tag() ) {
-		return $image;
-	}
-
-	$width             = $image_elm->get_attribute( 'width' );
-	$height            = $image_elm->get_attribute( 'height' );
-	$loading_val       = $image_elm->get_attribute( 'loading' );
-	$fetchpriority_val = $image_elm->get_attribute( 'fetchpriority' );
+	$width             = preg_match( '/ width="([0-9]+)"/', $image, $match_width ) ? (int) $match_width[1] : null;
+	$height            = preg_match( '/ height="([0-9]+)"/', $image, $match_height ) ? (int) $match_height[1] : null;
+	$loading_val       = preg_match( '/ loading="([A-Za-z]+)"/', $image, $match_loading ) ? $match_loading[1] : null;
+	$fetchpriority_val = preg_match( '/ fetchpriority="([A-Za-z]+)"/', $image, $match_fetchpriority ) ? $match_fetchpriority[1] : null;
```

Comment:
@kt-12 Strictly speaking, we need to also cater for the case of single quotes around attributes which, while not common, is possible to be used, so basically use `"|'` instead of `"` in a few places.

Reply:
@felixarntz I have added support for both single and double quotes.